### ISSUE TITLE
React 0.14

### ISSUE THREAD

**Issue by mhahn** [casesandberg/reactcss#22]
Is the use of `is="..."` going to break with React 0.14?

http://facebook.github.io/react/blog/2015/09/10/react-v0.14-rc1.html#notable-enhancements

"is" is listed as a support HTML attribute now.

**Comment by casesandberg** [comment]
Do you know what the `is="..."` attribute does? I cant find anything about it. 

Also, since the `react-map-styles` is doing string replace, it wont break anything unless you are trying to use `is`. I have been trying to come up with a more long-term solution other than hijacking it anyways.

**Comment by mhahn** [comment]
no idea, i just was reading the new react documentation and thought of
this.

i meant more of "conflict" than break. i'll see if i can lookup anything
about "is"

**Comment by seekshiva** [comment]
There is another issue related to React 0.14. Should I post it here or create a new issue?

**Comment by seekshiva** [comment]
Seems like react is enforcing that all react components extend React.Component directly.

```
class MyComponent extends ReactCSS.Component {
  render(){
    return <div />
  }
}
```

throws this warning:

```
Warning: MyComponent(...): React component classes must extend React.Component
```

**Comment by casesandberg** [comment]
Found out that the `is="..."` syntax is for web component support and not recommended by Facebook so we should be fine on that front.

https://twitter.com/soprano/status/651880693748899840

**Comment by casesandberg** [comment]
Looking into the component classes error as we speak.

**Comment by jhytonen** [comment]
Don't know if the component classes issue is really an issue?

The warnings appear, because reactcss has a dependency to react 0.13. That version does not have "isReactComponent" flag set for the React.Component class from which ReactCSS.Component inherits. When I updated reactcss' dependecy to react 0.14, the warnings disappeared and everything works just fine.

Here is the code that React uses to validate the inheritance:
`warning(Component.prototype && Component.prototype.isReactComponent || !canInstantiate || !(inst instanceof Component), '%s(...): React component classes must extend React.Component.', Component.displayName || Component.name || 'Component')`

**Comment by mhahn** [comment]
@casesandberg sorry for blending this and https://github.com/casesandberg/reactcss/issues/23 together. @jhytonen is right, I upgraded to the version on master and the warnings etc. have gone away. everything seems to be working well.

**Comment by jhytonen** [comment]
@casesandberg Would it be possible for you to release a version of reactcss with the dependency fixed?

**Comment by casesandberg** [comment]
Dependency has been fixed in `reactcss@0.4.2`
